app/frontend/src/features/profile/ProfileTagInput.tsx

```diff
@@ -0,0 +1,223 @@
+      zIndex: 1,
+      color: "#586069",
+      backgroundColor: "#f6f8fa",
+    },
+    button: {
+      borderRadius: theme.shape.borderRadius,
+      borderColor: "rgba(0, 0, 0, 0.23)",
+      borderWidth: 1,
+      borderStyle: "solid",
+      padding: "18.5px 14px",
+      margin: theme.spacing(1, 0),
+      width: "100%",
+      fontSize: "16px",
+      justifyContent: "space-between",
+    },
+    tag: {
+      padding: "0 14px",
+      margin: theme.spacing(1, 0),
+      fontSize: "16px",
+      display: "flex",
+      alignItems: "center",
+    },
+    tagLabel: {
+      marginLeft: theme.spacing(1),
+    },
+    header: {
+      borderBottom: "1px solid #e1e4e8",
+      padding: "8px 10px",
+      fontWeight: 600,
+    },
+    inputBase: {
+      padding: 10,
+      width: "100%",
+      borderBottom: "1px solid #dfe2e5",
+      "& input": {
+        borderRadius: 4,
+        backgroundColor: theme.palette.common.white,
+        padding: 8,
+        transition: theme.transitions.create(["border-color", "box-shadow"]),
+        border: "1px solid #ced4da",
+        fontSize: 14,
+        "&:focus": {
+          boxShadow: `${fade(theme.palette.primary.main, 0.25)} 0 0 0 0.2rem`,
+          borderColor: theme.palette.primary.main,
+        },
+      },
+    },
+    paper: {
+      boxShadow: "none",
+      margin: 0,
+      color: "#586069",
+    },
+    option: {
+      minHeight: "auto",
+      alignItems: "flex-start",
+      padding: 8,
+      '&[aria-selected="true"]': {
+        backgroundColor: "transparent",
+      },
+      '&[data-focus="true"]': {
+        backgroundColor: theme.palette.action.hover,
+      },
+    },
+    popperDisablePortal: {
+      position: "relative",
+    },
+    checkbox: {
+      padding: 0,
+      marginRight: theme.spacing(1),
+    },
+  })
+);
+
+interface ProfileTagInputProps {
+  onChange: (_: any, value: string[]) => void;
+  value: string[];
+  options: string[];
+  label: string;
+}
+
+export default function ProfileTagInput({
+  onChange,
+  value,
+  options,
+  label,
+}: ProfileTagInputProps) {
+  const classes = useStyles();
+
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const [valueState, setValueState] = useState<string[]>(value);
+  const [pendingValue, setPendingValue] = useState<string[]>([]);
+
+  useEffect(() => {
+    setValueState(value);
+  }, [value]);
+
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setPendingValue(value);
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = (
+    event: React.ChangeEvent<{}>,
+    reason: AutocompleteCloseReason
+  ) => {
+    if (reason === "toggleInput") {
+      return;
+    }
+    onChange(null, pendingValue);
+    if (anchorEl) {
+      anchorEl.focus();
+    }
+    setAnchorEl(null);
+  };
+
+  const handleRemove = (tag: string) => {
+    onChange(
+      null,
+      value.filter((v) => v !== tag)
+    );
+  };
+
+  const open = Boolean(anchorEl);
```

Comment:
A shorter way to do a conversion to boolean is:
```suggestion
  const open = !!anchorEl;
```
to number: `+v` instead of Number(v)
to string: `''+v` instead of String(v)

I once worked where those shorter versions were preferred. Maybe we should not care too much about it?

Reply:
Not `!!anchorEl` instead to convert to boolean and preserve its original truthy value?

Reply:
Indeed, I meant a conversion to boolean.